test/test_torch.py

```diff
@@ -4703,11 +4703,11 @@ def test_error_msg_type_translation(self):
                 # message includes both Double and Long
                 '(?=.*Double)(?=.*Long)'):
 
-            # Calls model with a DoubleTensor input but LongTensor weights
-            input = torch.autograd.Variable(torch.randn(1, 1, 1, 6).double())
-            weight = torch.zeros(1, 1, 1, 3).long()
+            # Calls model with a LongTensor input but DoubleTensor weights
```

Comment:
Why did you invert the logic here?
Why keep input as a DoubleTensor and weight a LongTensor?

Reply:
I changed the test to remove `.data`, but I realized after looking at it again that this is somewhat unnecessary as the use of `.data` here is to access the tensor inside the parameter. However, I did have a reason for making the change. In particular,
* Assigning `model.weight = weight` is only allowed if `weight` is a parameter.
* If `weight` is to be a parameter, it can't be a LongTensor.

Using `model.weight.data = weight` allow us to skip the type check, but it doesn't seem like that is intended to be the tested outcome. This made the change of types necessary to remove `.data`. However, as this is using a different method of `.data`, I can revert this test.

Reply:
Ho right we have checks for Parameters that won't allow you to do this.
You can leave it as is.

Reply:
I don't think this is the right change to make.  This is a case (as you correctly pointed out) that using `.data` allows us to get around type checks.  So we really want to be testing `.data` here.